src/main/java/io/stargate/sgv2/jsonapi/service/shredding/model/WritableShreddedDocument.java

```diff
@@ -65,6 +68,7 @@ public static class Builder implements ShredListener {
     private Map<JsonPath, Boolean> queryBoolValues;
     private Map<JsonPath, BigDecimal> queryNumberValues;
     private Map<JsonPath, String> queryTextValues;
+    private Map<JsonPath, Date> queryTimestampValues;
```

Comment:
Are we not going to keep it java.time.Instant?

Reply:
We can choose that if it makes more sense -- I don't have strong preference. All we need is a wrapper for 64-bit `long`. Let me read a bit about `java.time.Instant`.

Reply:
Ok, as per JDK Javadocs:

"The range of an instant requires the storage of a number larger than a long. To achieve this, the class stores a long representing epoch-seconds and an int representing nanosecond-of-second, which will always be between 0 and 999,999,999. "

so I think that for our purposes it is more efficient to use otherwise outdated `java.util.Date`; that uses less memory.

I am ok going with alternatives (including `java.time.Instant`) if we find reasons to do so, but I think for now will go with `java.util.Date` just as a light-weight wrapper.

Reply:
I am for `java.time.Instant` as well.. Date is outdated :smile:

Reply:
That s what I pointed out in one of the comments, maybe we should simply operate on long values.

Reply:
Agree. @tatu-at-datastax Can we have this as just `long`?

Reply:
I think `Date` is better for type-safety (as sort of documentation), over wrapper type `Long`.
I guess it is possible to document things with naming conventions if we really want to do it, but I am not sure there is much benefit from doing that. To me `java.util.Date` seems like a convenient abstraction to use here.

If we really want to change this, sure, it can be done, but I don't think it greatly matters.